src/SideNavigation/SideNavigation.js

```diff
@@ -1,204 +1,286 @@
-                                                            key={ch.id}
-                                                            onClick={e => this.handleSelectChild(e, ch.id)}
-                                                            to={{ pathname: ch.link }}>
-                                                            {ch.name}
-                                                        </Link>
-                                                    ) : null}
-
-                                                    {ch.url ? (
-                                                        <a
-                                                            className={this.getSubLinkClasses(ch.id)}
-                                                            href={ch.url}
-                                                            key={ch.id}
-                                                            onClick={e => this.handleSelectChild(e, ch.id)}>
-                                                            {ch.name}
-                                                        </a>
-                                                    ) : null}
-                                                </React.Fragment>
-                                            );
-                                        })}
-                                    </ul>
-                                ) : null}
-                            </li>
-                        );
-                    })}
-                </ul>
-            </BrowserRouter>
+            <nav {...rest} className={sideNavClasses}>
+                {React.Children.map(children, (child) => {
+                    if (React.isValidElement(child)) {
+                        return React.cloneElement(child, {
+                            expandedIds: this.state.expandedIds,
+                            onItemSelect: this.handleSelect,
+                            selectedId: this.state.selectedId
+                        });
+                    } else {
+                        return child;
+                    }
+                })}
+            </nav>
         );
     }
 }
-SideNavList.propTypes = {
-    items: PropTypes.array.isRequired,
-    className: PropTypes.string
-};
 
-SideNavList.propDescriptions = {
-    items: 'An array of objects with keys \'id\', \'url\', \'name\', \'hasChild\', \'child\', and \'glyph\' setting the attributes of the items.'
+SideNav.propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    icons: PropTypes.bool,
+    selectedId: PropTypes.string
 };
 
-export const SideNavGroup = props => {
-    const { title, children, className, titleProps, ...rest } = props;
+SideNav.propDescriptions = {
+    icons: 'Set to **true** enables side navigation collapsed with icons.',
+    selectedId: 'The id of the selected `SideNavItem`.'
+};
 
+export const SideNavGroup = ({children, className, expandedIds, onItemSelect, selectedId, title, titleProps, ...props}) => {
```

Comment:
Lets remove `SideNavGroup`, and add title prop to `SideNavList`

Reply:
Done!